src/sentry/search/eap/resolver.py

```diff
@@ -77,6 +77,28 @@ class SearchResolver:
         ],
     ] = field(default_factory=dict)
 
+    def get_function_definition(
+        self, function_name: str
+    ) -> ConditionalAggregateDefinition | FormulaDefinition | AggregateDefinition:
+        if function_name in self.definitions.aggregates:
+            return self.definitions.aggregates[function_name]
+        elif function_name in self.definitions.formulas:
+            return self.definitions.formulas[function_name]
+        elif function_name in self.definitions.conditional_aggregates:
+            return self.definitions.conditional_aggregates[function_name]
+        else:
+            raise InvalidSearchQuery(f"Unknown function {function_name}")
+
+    def check_if_function_is_enabled(self, function_name: str) -> bool:
+        function_definition = self.get_function_definition(function_name)
+        if not function_definition.private:
+            return True
+
+        if function_name in self.config.functions_acl:
+            return True
+
+        return False
```

Comment:
nit: but I find this far more easy to read, a function is disabled if its private and not in the ACL
```suggestion
    def check_if_function_is_disabled(self, function_name: str) -> bool:
        function_definition = self.get_function_definition(function_name)
        return function_definition.private and function_name not in self.config.functions_acl
```